File: src/webcore/mvc/model/abstractmodel.ts
"use strict";

export class AbstractModel {
	public _wasm_model: any = null;

	constructor(context: any) {
		this._wasm_model = new context.module.AbstractModel();
	}

	public getProperties(): any {
		return this._wasm_model.getProperties();
	}

	public setProperty(key, value): void {
		this._wasm_model.setProperty(key, value);
	}
};

File: src/webcore/mvc/model/abstractmodel.cpp
#include <sway/webcore/mvc/model/abstractmodel.h>

NAMESPACE_BEGIN(sway)
NAMESPACE_BEGIN(webcore)
NAMESPACE_BEGIN(mvc)
NAMESPACE_BEGIN(model)

ModelSmartPtr_t AbstractModel::create() {
	return std::make_shared<AbstractModel>();
}

AbstractModel::AbstractModel()
	: _data(emscripten::val::object()) {
	setProperty("id", "0");
}

AbstractModel::~AbstractModel() {
	// Empty
}

void AbstractModel::setProperty(const std::string & key, const std::string & value) {
	_data.set(key, emscripten::val(value));
	notify();
}

emscripten::val AbstractModel::getProperties() const {
	return _data;
}

NAMESPACE_END(model)
NAMESPACE_END(mvc)
NAMESPACE_END(webcore)
NAMESPACE_END(sway)
